src/main/java/org/icgc_argo/workflow/search/graphql/fetchers/EntityDataFetchers.java

```diff
@@ -62,9 +62,9 @@ public AsyncDataFetcher getDataFetcher() {
                   }
                   if (ANALYSIS_ENTITY.equals(values.get("__typename"))) {
                     final Object analysisId = values.get("analysisId");
-                    final Object studyId = values.get("studyId");
-                    if (analysisId instanceof String && studyId instanceof String) {
-                      return Mono.just(new Analysis((String) analysisId, (String) studyId));
+                      final Object studyId = values.get("studyId");
+                    if (analysisId instanceof String || studyId instanceof String) {
+                      return Mono.just(new Analysis((String) analysisId, (String) studyId);
```

Comment:
Just confirming before approval, will this handle properly `null` values for analysisId and studyId?

Reply:
Good point, I have updated the Analysis.java constructor logic to check the null values.